src/OpenTelemetry.Exporter.Prometheus.HttpListener/PrometheusHttpListener.cs

```diff
@@ -105,13 +114,23 @@ public void Stop()
             }
         }
 
-        /// <inheritdoc/>
         public void Dispose()
         {
-            if (this.httpListener != null && this.httpListener.IsListening)
+            this.Dispose(true);
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (!this.disposed)
             {
-                this.Stop();
-                this.httpListener.Close();
+                if (disposing && this.httpListener != null && this.httpListener.IsListening)
+                {
+                    this.Stop();
+                    this.httpListener.Close();
+                    this?.Dispose();
```

Comment:
Is the null conditional necessary here?

Reply:
Nope, I've removed the null check for this.Dispose(). Thanks!